File: contrib/pyre/tests/libpyre/journal/firewall-injection.cc
// -*- coding: utf-8 -*-
//
// michael a.g. aïvázis
// orthologue
// (c) 1998-2019 all rights reserved
//


// for the build system
#include <portinfo>

// packages
#include <cassert>
// access to the journal header file
#include <pyre/journal.h>

// main program
int main() {

    // instantiate a firewall channel
    pyre::journal::firewall_t firewall("pyre.journal.test");
    firewall.deactivate();

    // inject all the standard manipulators and built in types
    firewall
        << pyre::journal::at(__HERE__)
        << pyre::journal::set("key", "value")
        << "Hello world!" << pyre::journal::newline
        << 0 << pyre::journal::newline
        << 0.0 << pyre::journal::endl;

    firewall
        << pyre::journal::at(__HERE__)
        << (void *)&firewall << pyre::journal::newline
        << std::string("Hello world!")
        << pyre::journal::endl;

    // all done
    return 0;
}

// end of file

File: contrib/pyre/tests/journal/firewall-injection.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis
# orthologue
# (c) 1998-2019 all rights reserved
#


"""
Verify that {firewall.log} works as expected
"""


def test():
    # access the package
    import journal
    # build a firewall channel
    firewall = journal.firewall("journal.test1")
    # deactivate it
    firewall.active = False

    # and make it say something
    firewall.log("hello world!")

    # all done
    return


# main
if __name__ == "__main__":
    test()


# end of file
